server/src/Infrastructure/Persistence/Repositories/Repository.cs

```diff
@@ -12,98 +14,70 @@ public abstract class Repository<TEntity> : IRepository<TEntity> where TEntity :
     public Action<TEntity> OnInstanceAdded = delegate { };
     public Action<TEntity> OnInstanceUpdated = delegate { };
 
-    private readonly string _filePath;
-    private readonly IFileService _fileService;
-    private readonly ApplicationDbContext _dbContext;
+    protected readonly ApplicationDbContext _dbContext;
 
-    protected Repository(string fileName, IFileService fileService, ApplicationDbContext dbContext)
+    protected Repository(ApplicationDbContext dbContext)
     {
-        _fileService = fileService;
-        _filePath = Path.Combine(
-            Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData),
-            "StudyZen",
-            $"{fileName}.json");
         _dbContext = dbContext;
 
         OnInstanceAdded += AuditableEntityInterceptor.SetCreateStamp;
         OnInstanceUpdated += AuditableEntityInterceptor.SetUpdateStamp;
     }
 
-    public void Add(TEntity instance)
+    public async Task Add(TEntity instance)
     {
-        var entitySet = GetEntitySet();
-
-        instance.Id = ++entitySet.TotalCount;
-        OnInstanceAdded(instance);
+        try
+        {
+            OnInstanceAdded(instance);
 
-        Log.Information("Adding {0} with id {1}", typeof(TEntity).Name, instance.Id);
-        entitySet.Instances.Add(instance);
+            await _dbContext.Set<TEntity>().AddAsync(instance);
+            await _dbContext.SaveChangesAsync();
+        }
+        catch (DbUpdateException ex) when ((ex.InnerException as SqlException)?.Number == 547)
+        {
+            throw new ValidationException("FK volation");
+        }
 
-        _fileService.WriteToJsonFile(_filePath, entitySet);
     }
 
-    public TEntity? GetById(int instanceId)
+    public async Task<TEntity?> GetById(int instanceId)
     {
-        return GetAll().FirstOrDefault(i => i.Id == instanceId);
+        var instance = await _dbContext.Set<TEntity>().FindAsync(instanceId);
+        return instance;
     }
 
-    public List<TEntity> GetAll()
+    public async Task<List<TEntity>> GetAll()
     {
-        return GetEntitySet().Instances;
+        return await _dbContext.Set<TEntity>().ToListAsync();
     }
 
-    public bool Update(TEntity instance)
+    public async Task<bool> Update(TEntity instance)
     {
-        var entitySet = GetEntitySet();
+        try
+        {
+            OnInstanceUpdated(instance);
 
-        var instanceIdx = entitySet.Instances.FindIndex(i => i.Id == instance.Id);
-        if (instanceIdx < 0)
+            _dbContext.Set<TEntity>().Attach(instance);
+            _dbContext.Entry(instance).State = EntityState.Modified;
+            await _dbContext.SaveChangesAsync();
+            return true;
+        }
+        catch (DbUpdateException)
```

Comment:
Maybe we should not catch exceptions at repository level? They would provide more context as to why operation failed rather than a boolean or nothing at all.

Additionally, not catching them would automatically mean a 5xx status instead of us guessing it at service.

Reply:
so then there is no point in having this method as bool

Reply:
Yep